addons/web/static/tests/boot_tests.js

```diff
@@ -1,18 +1,66 @@
-(function () {
+(async function () {
 "use strict";
 
-odoo.__DEBUG__.didLogInfo.then(function () {
-    var modulesInfo = odoo.__DEBUG__.jsModules;
+    // We define here a test asserting that no module failed to be loaded. The
+    // module system (see boot.js) waits 10 seconds before logging that some
+    // modules couldn't be loaded, when it happens. The test below is really
+    // important as it will detect testing modules that can't be loaded, for
+    // instance because one of their dependencies is missing. Without it, we
+    // could have tests that aren't executed, without any notice.
+    //
+    // We have the following constraints around this test:
+    //  - we want it to be always executed (obviously)
+    //  - we need it to wait for the module system to complete the loading of
+    //    the modules before being executed (wait for the didLogInfo promise)
+    //  - we don't want it to slow down the whole suite
```

Comment:
I would be in favor of relaxing this condition if it can make the whole thing 100% deterministic, related to my other comments.

Reply:
it will be super annoying as this test will always be executed first